packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/SyncedRealmTests.kt

```diff
@@ -1547,6 +1552,39 @@ class SyncedRealmTests {
         }
     }
 
+    @Test
+    fun cannotSyncCollectionsInMixed() = runBlocking {
+        val flexApp = TestApp(
+            logLevel = LogLevel.ALL,
+            appName = io.realm.kotlin.test.mongodb.TEST_APP_FLEX,
+            builder = {
+                it.syncRootDirectory(PlatformUtils.createTempDir("flx-sync-"))
+            }
+        )
+        val (email, password) = randomEmail() to "password1234"
+        val user = flexApp.createUserAndLogIn(email, password)
+        val local = createFlexibleSyncConfig(user = user, name = "local", schema = setOf(JsonStyleRealmObject::class)) {
+            initialSubscriptions {
+                this.add(it.query<JsonStyleRealmObject>())
+            }
+        }
+        Realm.open(local).use {
+            it.write {
+                val obj = copyToRealm(JsonStyleRealmObject())
+                assertFailsWithMessage<IllegalStateException>("Cannot sync nested set") {
+                    obj.value = realmAnySetOf()
+                }
+                assertFailsWithMessage<IllegalStateException>("Cannot sync nested list") {
+                    obj.value = realmAnyListOf()
+                }
+                assertFailsWithMessage<IllegalStateException>("Cannot sync nested dictionary") {
+                    obj.value = realmAnyDictionaryOf()
+                }
+            }
+        }
+        flexApp.close()
```

Comment:
We should use the new `use` construct, otherwise the app will leak if a test fails before hitting this.